src/main/java/io/stargate/sgv2/jsonapi/api/model/command/deserializers/FilterClauseDeserializer.java

```diff
@@ -230,6 +230,11 @@ private void validate(
                   ErrorCode.INVALID_FILTER_EXPRESSION,
                   "$size operator must have integer value >= 0");
             }
+            // Check if the value is an integer by comparing its scale.
+            if (i.stripTrailingZeros().scale() > 0) {
+              throw new JsonApiException(
+                  ErrorCode.INVALID_FILTER_EXPRESSION, "$size operator must have an integer value");
```

Comment:
Why this is needed, the previous line validation should have taken care of this?

Reply:
This is for user inputs that are not Integer.

{
    "find": {
        "filter": {
                "ages": {
                    "$size":  2.43
                }
        }
    }
}